Fix with oscdimg.exe search, ExecutionPolicy, compression, etc..

Snshadow opened the pull request:
- Edit README.md part for `Set-ExecutionPolicy` to use `-Scope Process` to prevent changing the entire computer's policy.
  - https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-5.1#:~:text=The%20Set%2DExecutionPolicy%20cmdlet%27s%20default%20scope%20is%20LocalMachine
- Fix tiny11maker.ps1 and tiny11Coremaker.ps1 to find oscdimg.exe from registry instead of using fixed path.
- Add Run.bat helper batch file(referenced from [Win11Debloat](https://github.com/Raphire/Win11Debloat)) to help running tiny11maker by double clicking the batch file from explorer.

Snshadow committed:
several-fixes

- Edit README.md part for `Set-ExecutionPolicy` to use `-Scope Process` to prevent changing the entire computer's policy.
- Fix tiny11maker.ps1 and tiny11Coremaker.ps1 to find oscdimg.exe from registry instead of using fixed path.
- Add Run.bat helper batch file(referenced from Win11Debloat) to help running tiny11maker with double clicking.

Snshadow commented:
Although it is stated that this project is open-source, it would be nice to add the license file as the github document says [However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work.](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository#:~:text=However%2C%20without%20a,open%20source%20license.)

If you've chosen not to license this project for some reason, I respect that decision 😊.

Snshadow committed:
fix: change disallowing ExecutionPolicy

- Add AllSigned, Undefined for consideration which also prevent script from running

Snshadow commented on tiny11Coremaker.ps1:
```diff
@@ -1,12 +1,14 @@
-# Check if PowerShell execution is restricted
-if ((Get-ExecutionPolicy) -eq 'Restricted') {
-    Write-Host "Your current PowerShell Execution Policy is set to Restricted, which prevents scripts from running. Do you want to change it to RemoteSigned? (yes/no)"
+# Check if PowerShell execution is Restricted or AllSigned or Undefined
```
I am curious if this part is actually called, is this for some possible cases where the user tries to run it from powershell they have been using for different activity which could have set ExecutionPolicy to the value that does not allow script to run? I think in normal cases, the user will run it with from the powershell which should have the ExecutionPolicy that allow to run custom scripts, and it is inherited, which make it will be never the ones that need to be changed for elevation.

Snshadow replied:
I am curious if this part is actually called, is this for some possible cases where the user tries to run it from powershell they have been using for different activity which could have set ExecutionPolicy to the value that does not allow script to run? I think in normal cases, the user will run it with from the powershell which should have the ExecutionPolicy that allow to run custom scripts, and it is inherited, which make it will be never the ones that need to be changed for elevation.

Snshadow committed:
fix: do not call method on $null object

- fix the cause of crash if ADK is not installed

Snshadow committed:
update: powershell dism feature and provisoned appx

- Get architecture using Get-WindowsImage and its property directly.
- Use Get-AppxProvisionedPackage and Remove-AppxProvisionedPackage instead if dism for more consistence.
- Use Export-WindowsImage with -CompressionType Maximum instead of Fast for smaller type(used to be recovery with dism.exe).

Snshadow committed:
refactor: use explict parameters

- align switch block

Snshadow committed:
enhance: reduce iso image size

- Revert to use dism for exporting install.esd to use recovery compression which reduced the size about 1 gigabytes.
- Add outlook provisioned appxpackage for removal.

Snshadow commented on tiny11maker.ps1:
```diff
@@ -387,10 +387,9 @@ Write-Host ' '
-# Compressiontype Recovery is not supported with PShell https://learn.microsoft.com/en-us/powershell/module/dism/export-windowsimage?view=windowsserver2022-ps#-compressiontype
-Export-WindowsImage -SourceImagePath $ScratchDisk\tiny11\sources\install.wim -SourceIndex $index -DestinationImagePath $ScratchDisk\tiny11\sources\install2.wim -CompressionType Maximum
+# Compressiontype Recovery is not supported with PShell https://learn.microsoft.com/en-us/powershell/module/dism/export-windowsimage?-ps#-compressiontype
+& dism /English /Export-Image "/SourceImageFile:$ScratchDisk\tiny11\sources\install.wim" "/SourceIndex:$index" "/DestinationImageFile:$ScratchDisk\tiny11\sources\install.esd" /Compress:recovery
```
From https://github.com/ntdevlabs/tiny11builder/commit/7ec6a7c253dda50255e795c2c481b3b93cdc5074, looks like using recovery compression to export wim image worked at some time? Currently when I tested with Window 11 23H2 and it creates garbage file which has nothing at all(this is somehow expected as [.esd extension is required for recovery compression](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism-image-management-command-line-options-s14?view=windows-11#export-image:~:text=Use%20the%20recovery%20option%20to%20export%20push%2Dbutton%20reset%20images.%20The%20resulting%20files%20are%20much%20smaller%20in%20size%2C%20which%20in%20turn%2C%20greatly%20reduce%20the%20amount%20of%20disk%20space%20needed%20for%20saving%20the%20push%2Dbutton%20reset%20image%20on%20a%20recovery%20drive.%20The%20destination%20file%20must%20be%20specified%20with%20an%20.esd%20extension.)(I somehow have a iso file which has wim file compressed with recovery extension(LZMS)(?), so it looks like it used to work previously at some time). Actually tiny11CoreMaker.ps1 currently export as .esd image, and it boots without issue, so I think keeping using dism in necessary for this part.

Karl-WE replied:
see https://github.com/ntdevlabs/tiny11builder/blob/af69a482650e6eb6542ae6fae358149bf8210fe3/tiny11maker.ps1#L391

https://github.com/ntdevlabs/tiny11builder/issues/317#issuecomment-2663329713

Snshadow committed:
refactor: add new line for more readable console lines

- revert packagePrefixed change as the package is already in github.com/ntdevlabs/tiny11builder/pull/253

Snshadow committed:
fix: look for WOW6432Node first

- registry value from WOW6432Node is regarded first for setup

journeym commented:
this definetly should be merged, at leatit works now with the reg and removal of software

Karl-WE reviewed (commented):
further comments

FREEWING-JP commented on Run.bat:
👍

xavierfingers replied:
ok

Loufe commented:
I would vote for accepting this PR, a lot of great small cleanups.

Snshadow committed:
fix: ScratchDisk variable and some more..

- fix (noted in https://github.com/ntdevlabs/tiny11builder/issues/339) by adding ':' colon for validation pattern, which used to be set to empty value(resulting in using script root path)
- use `Export-WindowsImage` with undocumented Compression Type for more consistence
- use direct string check for little performance improvement

Snshadow commented on tiny11maker.ps1:
```diff
@@ -388,10 +386,9 @@ Write-Host ' '
-Remove-Item -Path "$ScratchDisk\tiny11\sources\install.wim" -Force | Out-Null
-Rename-Item -Path "$ScratchDisk\tiny11\sources\install2.wim" -NewName "install.wim" | Out-Null
+# Run `Export-WindowsImage` with undocumented CompressionType "LZMS" (which is the same compression used for Recovery from dism.exe)
+Export-WindowsImage -SourceImagePath "$ScratchDisk\tiny11\sources\install.wim" -SourceIndex "$index" -DestinationImagePath "$ScratchDisk\tiny11\sources\install2.wim" -CompressionType "LZMS"
```
Although it is said that [Compressiontype Recovery is not supported with PShell](https://learn.microsoft.com/en-us/powershell/module/dism/export-windowsimage?-ps#-compressiontype), actually `Export-WindowsImage` used to support the same compression algorithm used for Recovery from Windows 10(it internally uses [WIMExportFile](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/wim/dd834952(v=msdn.10)), which used to support [LZMS compression](https://github.com/jeffkl/ManagedWimgApi/blob/1b94dad150f5994e586d7ca4c32dddf967c7eed1/wimgapi.h#L37)) and it even works with .wim extension.

Karl-WE reviewed (commented):
It has been noted in comments that Export Image doesn't support compress which is really odd.

Thanks for your improvements. Unfortunately there are many open PRs that would conflict with your PR.
I hope that the owner can find some time to find a clean way to merge all pending PRs, each of them have unique improvements.

Snshadow committed:
Merge ntdevlabs/tiny11builder/main into several-fixes

Snshadow committed:
fix: Instruction list number

xavierfingers commented:
This should be merged
